projects/make-a-vis/src/lib/data-view/shared/data.service.ts

```diff
@@ -70,18 +100,106 @@ export class DataService {
       });
   }
 
+  /**
```

Comment:
This feels like a lot of code to do a simple thing. getDisplayValue, checkSize, getTooltip, could easily be removed with a couple more lines in injectDisplayVariables.